nebula_ros/src/aeva/hw_monitor_wrapper.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 TIER IV, Inc.
+
+#include "nebula_ros/aeva/hw_monitor_wrapper.hpp"
+
+#include "nebula_ros/common/parameter_descriptors.hpp"
+
+#include <nebula_common/aeva/packet_types.hpp>
+#include <nebula_common/util/parsing.hpp>
+#include <nlohmann/json_fwd.hpp>
+
+#include <diagnostic_msgs/msg/detail/diagnostic_array__builder.hpp>
+#include <diagnostic_msgs/msg/detail/diagnostic_array__struct.hpp>
+#include <diagnostic_msgs/msg/detail/diagnostic_status__struct.hpp>
+#include <diagnostic_msgs/msg/detail/key_value__struct.hpp>
+
+#include <algorithm>
+#include <chrono>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace nebula::ros
+{
+
+using diagnostic_msgs::msg::DiagnosticStatus;
+using drivers::aeva::HealthCode;
+using nebula::util::get_if_exists;
+
+AevaHwMonitorWrapper::AevaHwMonitorWrapper(
+  rclcpp::Node * const parent_node, std::shared_ptr<const drivers::aeva::Aeries2Config> config)
+: logger_(parent_node->get_logger().get_child("HwMonitor")),
+  parent_node_(parent_node),
+  config_(std::move(config))
+{
+  diag_span_ = parent_node->declare_parameter<uint16_t>("diag_span", 500, param_read_only());
+
+  diagnostics_pub_ =
+    parent_node->create_publisher<diagnostic_msgs::msg::DiagnosticArray>("/diagnostics", 10);
```

Comment:
```suggestion
    parent_node->create_publisher<diagnostic_msgs::msg::DiagnosticArray>("diagnostics", 10);
```

Reply:
Addressed in [5a8eea1](https://github.com/tier4/nebula/pull/169/commits/5a8eea197609d22dd8e4c312c66610b3bd8c3c72)